### ISSUE TITLE
Getting errors after IEC replaced the meter

### ISSUE THREAD

**Issue by chu11u** [GuyKh/iec-custom-component#223]
## System Information

version | core-2024.11.1
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.4
os_name | Linux
os_version | 6.6.54-haos
arch | x86_64
timezone | Asia/Jerusalem
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
HACS Data | ok
GitHub API Calls Remaining | 5000
Installed Version | 2.0.1
Stage | running
Available Repositories | 1542
Downloaded Repositories | 13

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | November 17, 2024 at 2:00 AM
relayer_connected | true
relayer_region | eu-central-1
remote_enabled | true
remote_connected | true
alexa_enabled | false
google_enabled | true
cloud_ice_servers_enabled | true
remote_server | eu-central-1-14.ui.nabu.casa
certificate_status | ready
instance_id | 7a2de663093148cd9a32928f9913aeb1
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 13.2
-- | --
update_channel | stable
supervisor_version | supervisor-2024.11.2
agent_version | 1.6.0
docker_version | 27.2.0
disk_total | 125.3 GB
disk_used | 45.0 GB
healthy | true
supported | true
host_connectivity | true
supervisor_connectivity | true
ntp_synchronized | true
virtualization | kvm
board | ova
supervisor_api | ok
version_api | ok
installed_addons | AirCast (4.2.2), AppDaemon (0.16.7), Dnsmasq (1.8.1), Double Take (1.13.1), Duck DNS (1.18.0), ESPHome (2024.10.3), ESPHome Device Compiler (dev) (2024.11.0-dev20241109), File editor (5.8.0), Filebrowser (2.23.0_14), Grocy (0.22.1), Home Assistant Google Drive Backup (0.112.1), Let's Encrypt (5.2.3), Mosquitto broker (6.4.1), NGINX Home Assistant SSL proxy (3.11.0), Node-RED (18.1.1), RTSPtoWeb - WebRTC (1.4.0), Samba share (12.3.2), Studio Code Server (5.17.3), TasmoAdmin (0.30.6), Terminal & SSH (9.15.0), eWeLink Smart Home (1.4.3), motionEye (0.21.0), Glances (0.21.1)

</details>

<details><summary>Dashboards</summary>

dashboards | 3
-- | --
resources | 6
views | 23
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | November 9, 2024 at 5:19 PM
-- | --
current_recorder_run | November 9, 2024 at 7:19 PM
estimated_db_size | 204.41 MiB
database_engine | sqlite
database_version | 3.45.3

</details>

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of any [previous issues](https://github.com/GuyKh/iec-custom-component/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

At first there was no change in the first few days. And now I get this error:

No devices or entities

Failed setup, will retry: object of type 'NoneType' has no len()

I am trying to understand if this as a result of the replacement and data will be presented in the future (according to IEC only after the first reading data will be first be available in their app and than every 15 minutes)
Or maybe this is a real problem

### Reproduction steps

Enable and disabled the integration
Removed and reconfigured the integration

### Debug logs

```text
Logger: custom_components.iec.coordinator
Source: helpers/update_coordinator.py:382
integration: Israel Electric Corporation (IEC) (documentation, issues)
First occurred: 9:44:24 AM (40 occurrences)
Last logged: 12:21:38 PM

Unexpected error fetching Iec data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 382, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/iec/coordinator.py", line 385, in _async_update_data
    kva_tariff = await self._get_kva_tariff()
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/iec/coordinator.py", line 189, in _get_kva_tariff
    self._kva_tariff = await self.api.get_kva_tariff()
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/iec_api/iec_client.py", line 511, in get_kva_tariff
    return await static_data.get_kva_tariff(self._session)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/iec_api/static_data.py", line 93, in get_kva_tariff
    await _get_tariffs(session)
  File "/usr/local/lib/python3.12/site-packages/iec_api/static_data.py", line 42, in _get_tariffs
    response = await commons.send_get_request(session=session, url=GET_KWH_TARIFF_URL)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/iec_api/commons.py", line 125, in send_get_request
    parse_error_response(resp, json_resp)
  File "/usr/local/lib/python3.12/site-packages/iec_api/commons.py", line 92, in parse_error_response
    if len(json_resp) > 0:
       ^^^^^^^^^^^^^^
TypeError: object of type 'NoneType' has no len()
```

**Comment by GuyKh** [comment]
@chu11u - it looks like you're using an old version. Can you check for update and try again?

**Comment by chu11u** [comment]
No update was detected so I manually re-downloaded it from HACS and restart. Now it is working, but still not detected the new meter. I guess it will update only after IEC will update on their side as well.

Thanks!!

**Comment by GuyKh** [comment]
Check first iec.co.il site for data, before expecting it to appear in HomeAssistant

**Comment by GuyKh** [comment]
Also - I fixed #128 
So expect this to rollout in the next version. (Please verify by setting logs to debug)
